### ISSUE TITLE
Request failed for https://p22-caldav.icloud.com returned code 301

### ISSUE THREAD

**Issue by michaelh82** [derekantrican/GAS-ICS-Sync#261]
I have been using this script successfully since November 2020.  However, upon opening my Google Calendar a couple of days ago, I noticed that all my Apple calendars have completely disappeared.

Upon going through the execution logs, I can see that the script started to randomly delete all the calendar events one evening and ever since, I am getting the following error:

`Error, Retrying...Exception: Request failed for https://p22-caldav.icloud.com returned code 301
`

I have checked the calendar share links from Apple and there are no changes there.  I have even re-entered the links into the GAS-ICS-Sync editor and tried re-installing/re-running the script, but the issue is still occurring.  I can also access the calendar just fine when I type the share link into a browser, so no issues with permissions.  Nothing else has changed, so I cannot seem to get my head around why it would suddenly delete everything and then loose connection.

If anyone can advise, that would be great 👍🏼

**Comment by derekantrican** [comment]
HTTP 301 is "Moved Permanently" and the response back is likely including the actual target should be in a response header called "Location". When you put the url in your browser, does it follow the redirect and take you to a different url?

**Comment by michaelh82** [comment]
Yeah, I was expecting to see a redirect as I was aware that 301 is moved permanently.  But, no redirect appears to happen.  When I visit the URL in my browser, Google Calendar just opens straight up and it tries to add the calendar.  Nothing seems to be any different, including the calendar URLs, so totally confused by it.

**Comment by jhaenen** [comment]
I ran into the same problem, it seems for me to fix it to disable "public calendar" of the calendar and renable it in icloud. You'll get a new URL that works flawlessly for me.
